lib/Transforms/UnusedMemRef/CMakeLists.txt

```
@@ -0,0 +1,20 @@
+set(LLVM_TARGET_DEFINITIONS UnusedMemref.td)
+mlir_tablegen(UnusedMemref.h.inc -gen-pass-decls -name UnusedMemRef)
+add_public_tablegen_target(MLIRHeirUnusedMemRefIncGen)
+
+add_mlir_dialect_library(MLIRHeirUnusedMemRef
+    PARTIAL_SOURCES_INTENDED
+    UnusedMemref.cpp
```

mismatched case causes build issues on linux
```suggestion
    UnusedMemRef.cpp
```